### PR TITLE
dynamic pagination for richlist WIP

### PR DESCRIPTION
**Pull request by rayanfer32**
## Description

- 

### Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor/Revamp ( rewrite or restructure code)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] I ran `npm run format`/`yarn format` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked my code and corrected any misspellings

### Related Issue

- none

### Proposed Changes

- none

### Additional Info

- none

### Screenshots

|              Original              |              Updated              |
| :--------------------------------: | :-------------------------------: |
|  |  |
|  |  |

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "editor.formatOnSave": true,
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "files.associations": {
-    "*.js": "javascript",
```
Comment (shrivatsabhat) [inline .vscode/settings.json:5]: Why js is removed?

Comment (rayanfer32) [reply .vscode/settings.json:5]: Its messing up the formatter

**Comment by rayanfer32** [comment]
Mostly an issue with how repl.it handles URLs , I have changed development URLs to the explorer.nexus.io API server.

File: src/components/Table/Table.jsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import { useTable, useSortBy, usePagination } from 'react-table';
 import styles from './Table.module.scss';
-import Pagination from './Pagination';
+// import Pagination from './Pagination';
```
Comment (shrivatsabhat) [inline src/components/Table/Table.jsx:4]: remove all unnecessary comments, keep it clean

File: src/components/UserAccount/TransactionDetails/TransactionDetails.jsx [hunk 3]
```diff
@@ -6,7 +6,8 @@ import CopyText from 'components/atoms/NE_CopyText/CopyText';
 import { useQuery } from 'react-query';
 import TYPES from 'types';
 import { useEffect, useState } from 'react';
-import DynamicPagination from 'components/Table/DynamicPagination';
+// import DynamicPagination from 'components/Table/DynamicPagination';
```
Comment (shrivatsabhat) [inline src/components/UserAccount/TransactionDetails/TransactionDetails.jsx:9]: remove comment

File: src/pages/blocks.js [hunk 4]
```diff
@@ -113,7 +123,10 @@ export default function Blocks(props) {
         <div style={{ overflow: 'visible' }}>
           <Table columns={columns} data={data} paginate={false} />
           <div style={{ marginBottom: '1rem' }}>
-            <DynamicPagination controls={dynamicPageControls} />
+            <DynamicPagination
+              controls={dynamicPageControls}
+              isStaticPanination={false}
```
Comment (shrivatsabhat) [inline src/pages/blocks.js:128]: block is it static pagination?

File: src/pages/transactions.js [hunk 5]
```diff
@@ -5,7 +5,8 @@ import { useQuery } from 'react-query';
 import Loader from 'components/atoms/NE_Loader';
 import Table from 'components/Table/Table';
 import { totalPages } from 'utils/helper';
-import DynamicPagination from 'components/Table/DynamicPagination';
+import DynamicPagination from 'components/atoms/NE_Pagination';
+// import DynamicPagination from 'components/Table/DynamicPagination';
```
Comment (shrivatsabhat) [inline src/pages/transactions.js:9]: comment

File: src/utils/common/debounce.js [hunk 6]
```diff
@@ -6,11 +6,6 @@
...
   let timeout;
-
-  return (...args) => {
-    clearTimeout(timeout);
-    timeout = setTimeout(() => {
-      cb(...args);
-    }, delay);
-  };
+  clearTimeout(timeout);
+  timeout = setTimeout(cb, delay);
```
Comment (shrivatsabhat) [inline src/utils/common/debounce.js:10]: hope this will not cause sideeffects

**Review by shrivatsabhat — APPROVED** [review]
please go through the comments once and proceed with merge

**Review by shrivatsabhat — APPROVED** [review]

**Pull request merged**